src/ApiCore/AuthWrapper.php

```diff
@@ -31,13 +31,24 @@
  */
 namespace Google\ApiCore;
 
+use DomainException;
+use Exception;
+use Google\Auth\ApplicationDefaultCredentials;
+use Google\Auth\Cache\MemoryCacheItemPool;
+use Google\Auth\FetchAuthTokenCache;
 use Google\Auth\FetchAuthTokenInterface;
+use Google\Auth\HttpHandler\Guzzle5HttpHandler;
+use Google\Auth\HttpHandler\Guzzle6HttpHandler;
+use Google\Auth\HttpHandler\HttpHandlerFactory;
+use Psr\Cache\CacheItemPoolInterface;
 
 /**
  * The AuthWrapper object provides a wrapper around a FetchAuthTokenInterface.
  */
 class AuthWrapper
 {
+    use ValidationTrait;
+
     private $fetchAuthTokenInterface;
```

Comment:
I'm not sure what you mean here - we accept a `FetchAuthTokenInterface` in the constructor

Reply:
Hmm ok, I guess I disagree but perhaps I am coming at it from a Java perspective, where this naming is normal (although it is not usual to have `Interface` in the name of the interface). Happy to change it, but then what should we call it, given that we can accept any implementing class?

Reply:
Re updating the name in the linked code, I agree the type should be `FetchAuthTokenInterface` instead of `CredentialsLoader`. That will be changing in the coming updates to constructor args, so we probably don't need to change it separately.

For the var name, what about `credentialsFetcher`? I wanted to not name it `credentialsLoader` specifically because it doesn't have to be a CredentialsLoader object.

Reply:
That sounds good! In fact I believe that is what we use in the handwritten code, so it would be great to use the same term.

Reply:
Updated